Tensile/Source/client/source/Reference.cpp

```diff
@@ -118,6 +118,38 @@ namespace Tensile
             throw std::runtime_error(msg.c_str());
         }
 
+        /* Transform the data type of A/B to Accumulator if A!=ACC or B!=ACC, but filter out cases, I8/I32/I32 and I8x4/I32/I32
+         * 
+         * There are three cases of doing multiplication and their conditions to do transform or not are as below.
+         * 1. AxB : (A!=ACC or B!=ACC) and A!=I8 and A!=I8x4
+         * 2. alpha x rC :  ((alpha!=ACC or rC!=ACC) and A!=I8 and A!=I8x4
+         * 3. Beta x D : (beta!=ACC or D!=ACC )and A!=I8 and A!=I8x4
+        */
+        template <typename A, typename B, typename Inputs, typename Accumulator>
+        constexpr bool need_transform
+            = !(std::is_same<A, Accumulator>() && std::is_same<B, Accumulator>())
+              && !std::is_same<typename Inputs::AType, Int8>()    //case I8/I32/I32
```

Comment:
Do we really need to consider "Inputs::AType"? Using "A" should be more reasonable and simple.

Reply:
The idea of using Inputs::AType is wanting to describe clearly that the filter cases are for two specific ContractionInputs. 

And, yes, we can use A instead Inputs::AType, which will get the same results as expected.
Since the value of A comes from
1. Matrix A
2. Alpha
3. Beta

And only case 1 which type has a chance to be I8 or I8x4.
I will use A instead Inputs::AType,